script.js

```diff
@@ -1,4 +1,275 @@
+};
+
+//advance game mode player2 auto chooses permutation
+var advanceMode2 = function () {
+  var counter2 = 0;
+  var findNumber = 6;
+  var convertPlayer2Results = "";
+  var storePlayer2FinalResults = 0;
+
+  while (findNumber > 0) {
+    while (counter2 < player2AdvanceRoll.length) {
+      if (player2AdvanceRoll[counter2] == findNumber) {
+        var store = player2AdvanceRoll[counter2];
+        convertPlayer2Results += store.toString();
+      }
+      counter2 += 1;
+    }
+    counter2 = 0;
+    findNumber -= 1;
+  }
+  storePlayer2FinalResults = parseInt(convertPlayer2Results);
+  finalPlayer2AdvancedResults = storePlayer2FinalResults;
+  myOutputValue = `Here are your dice rolls, we've permutated them accordingly to form the following numbers: ${storePlayer2FinalResults}`;
+  return myOutputValue;
+};
+
+// Winning message for advanced
+var winMsgAdvanced = function () {
+  if (finalPlayer1AdvancedResults > finalPlayer2AdvancedResults) {
+    player1WinCount += 1;
+    myOutputValue = `Player 1 wins this round! So far, Player 1 has won ${player1WinCount} and Player 2 has won ${player2WinCount}. Click submit to play again!`;
+    currentGameMode = "";
+    return myOutputValue;
+  }
+  player2WinCount += 1;
+  myOutputValue = `Player 1 wins this round! So far, Player 1 has won ${player1WinCount} and Player 2 has won ${player2WinCount} Click submit to play again!`;
+  currentGameMode = "";
+  return myOutputValue;
+};
+
+// Winning message
+var winMsg = function () {
+  if (player1FinalDiceRollResult > player2FinalDiceRollResult) {
+    player1WinCount += 1;
+    myOutputValue = `Player 1 wins this round! So far, Player 1 has won ${player1WinCount} and Player 2 has won ${player2WinCount}. Click submit to play again!`;
+    currentGameMode = "";
+    return myOutputValue;
+  }
+  player2WinCount += 1;
+  myOutputValue = `Player 1 wins this round! So far, Player 1 has won ${player1WinCount} and Player 2 has won ${player2WinCount} Click submit to play again!`;
+  currentGameMode = "";
+  return myOutputValue;
+};
+
+//RandomNumber generator 1 to 6
+var rollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var randomNumberResult = randomInteger + 1;
+  return randomNumberResult;
+};
+
+//Player 1 two dice rolls msgs
+var play1 = function () {
+  var diceRoll1 = player1DiceRoll1;
+  var diceRoll2 = player1DiceRoll2;
+  myOutputValue = `You have rolled ${diceRoll1} for Dice 1 and ${diceRoll2} for Dice 2!`;
+  return myOutputValue;
+};
+
+//Player 2 two dice rolls msgs
+var play2 = function () {
+  var diceRoll1 = player2DiceRoll1;
+  var diceRoll2 = player2DiceRoll2;
+  myOutputValue = `You have rolled ${diceRoll1} for Dice 1 and ${diceRoll2} for Dice 2!`;
+  return myOutputValue;
+};
+
+//Choose order Msg
+var chooseOrderMsg = function () {
+  myOutputValue = "Choose the order of your die by inputting either '1' or '2'";
+  return myOutputValue;
+};
```

Comment:
Functions typically carry out tasks and calculations. If its only outputting 1 statement, it isn't necessary to have a function.